crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -111,6 +111,30 @@ impl AccountInterface {
         &self.components
     }
 
+    /// Returns the authentication scheme used by this account, if any.
+    ///
+    /// This method searches through the account's component interfaces to find an authentication
+    /// component and returns its associated authentication scheme.
+    ///
+    /// # Returns
+    /// * `Some(AuthScheme)` - If the account has an authentication component
+    /// * `None` - If the account has no authentication components
+    ///
+    /// # Note
+    /// If the account has multiple authentication components, this method returns the first one
+    /// found.
+    pub fn get_auth_scheme(
+        &self,
+        storage: &miden_objects::account::AccountStorage,
+    ) -> Option<AuthScheme> {
+        for component in &self.components {
+            if let Some(auth_scheme) = component.get_auth_scheme(storage) {
+                return Some(auth_scheme);
+            }
+        }
+        None
+    }
```

Comment:
I'm not sure this method is correct and/or needed. We already have `Self::auth()` method which I think accomplishes the same thing.